docs/advanced-features/tabulator.md

```diff
@@ -0,0 +1,127 @@
+
+## Configuration
+
+Each Tabulator configuration is written in YAML, following the structure
+outlined below.
+
+```yaml
+
+### Example
+
+```yaml
+schema:
+  - name: name  # must match ^[a-z_][a-z0-9_]*$
+    type: STRING  # usually BOOLEAN, INT, FLOAT, DOUBLE, STRING, DATE, TIMESTAMP
+  - name: length
+    type: INT
+  - name: effective_length
+    type: FLOAT
+  - name: tpm
+    type: FLOAT
+  - name: num_reads
+    type: FLOAT
+source:
+  type: quilt-packages  # currently the only supported type
+  package_name: "^ccle/"
+  logical_key: "salmon/(?<sample_id>[^/]+)/quant*\\.genes\\.sf$"
+parser:
+  format: csv  # or `parquet`
+  delimiter: "\t"
+  header: true
+```
+
+1. **Schema**: The schema defines the columns in the table. Each column must
+   have a name and a type. The name must match the regular expression
+   `^[a-z_][a-z0-9_]*$`.  It does not need to match the column names in the
+   document.  However, if the column names are present in the document, you must
+   set `header` to `true` in the parser configuration.
+2. **Types**:  Must be uppercase and match the [Apache Arrow Data
+Types](https://github.com/awslabs/aws-athena-query-federation/tree/master/athena-federation-sdk#datatypes)
+used by Amazon Athena.  Valid types are BOOLEAN, TINYINT, SMALLINT, INT, BIGINT,
+FLOAT, DOUBLE, STRING, BINARY, DATE, TIMESTAMP.
+3. **Source**: The source defines the packages and objects to query. The `type`
+   must be `quilt-packages`. The `package_name` is a regular expression that
+   matches the package names to include. The `logical_key` is a regular
+   expression that matches the keys of the objects to include. The regular
+   expression may include a named capture group that will be added as a column
+   to the table.
+4. **Parser**: The parser defines how to read the files. The `format` must be
+   one of `csv` or `parquet`. The optional `delimiter` is the character used to
+   separate fields in the CSV file. The optional `header` field is a boolean
+   that indicates whether the first row of the CSV file contains column names.
+
+### Added columns
+
+In addition to the columns defined in the schema, Tabulator will add:
+
+- any named capture groups from the logical key regular expression
+- `$pkg_name` for the package name
+- `$logical_key` for the object as referenced by the package
+- `$physical_key` for the underlying S3 key
+- `$top_hash` for the revision of the package containing the object (which is
+  currently always `latest`)
+
+### Caveats
+
+1. **Schema Consistency**: All files in the package that match the logical key
+   must have the same schema as defined in the configuration.
+2. **Memory Usage**: Tabulator currently reads each file into memory, which may
+   cause issues with very large files. Additionally, Athena has a 16MB limit per
+   row.
+3. **Cost Management**: Querying very large datasets can be expensive
+   (approximately dollars per terabyte). Be sure to set up appropriate cost
```

Comment:
$5 by Athena and around $1 for lambda
`WHERE` and `LIMIT` may decrease amount of data processed by Athena, but doesn't affect that for lambda
omitting columns decreases amount of data processed by Athena and may also affect that for lambda
(I'm not sure that should end up in the doc but just in case)

Reply:
Good to know.